### ISSUE TITLE
Specified VAF does not work

### ISSUE THREAD

**Issue by LucieCCBC** [adamewing/bamsurgeon#124]
Hello!

When I run addsnv.py with a specified BED file, see table, it generates a wrong output. There are no errors but the file doesn't seem right, see figure for output result. 

![image](https://user-images.githubusercontent.com/44841082/48136964-42047300-e2a1-11e8-81c5-8379b3a59766.png)

You can see that I specify the VAF 1.0 but when I checked the generated BAM file in IGV, it gives a VAF 0.5 back, see picture below. 

![image](https://user-images.githubusercontent.com/44841082/48137126-a7586400-e2a1-11e8-820c-37d3792f8458.png)

So in short, I set the VAF on 1.0 but the output generates a VAF of 0.5.
How is this possible? 
 
Thanks,

Lucie

**Comment by adamewing** [comment]
Sorry to hear you're not seeing the expected output. I need a bit more information to detemine what's going on. Would you be able to send me a log generated as follows:

`addsnv.py (...options...) > LucieCCBC.log.txt 2>&1`

And either attach it to your reply or email to adam.ewing@gmail.com if you'd rather it not be public.

**Comment by LucieCCBC** [comment]
addsnv.py -v transporters.bed -f RMNISTHS.bam -r hg19.fa -o Output_RMNISTHS > LucieCCBC.log.txt 2>&1 

Gives contig mismatch: MT and Target and donor are aligned to incompatable reference genomes, see document. 
[LucieCCBC.log.txt](https://github.com/adamewing/bamsurgeon/files/2561239/LucieCCBC.log.txt)


That MT mismatch is logical because the added .bed file only contains transport regions, so MT is missing.

**Comment by adamewing** [comment]
So the reason you're seeing the VAF decrease for 10_101604107 is because you've got mutations at 10_101604130, 10_101604162, etc which would share the same reads. Mutations either need to be spaced apart such that they don't share reads, or use the `-z` option to link them (which will mean they are forced to share a VAF). This limitation exists because addsnv considers each mutation in isolation unless -z is specified (see  issue #123).

Regarding the incompatible reference genomes, the input BAM and the reference need to contain the same chromosomes and contigs. As a potential workaround, you can try the script in bamsurgeon/scripts/match_fasta_to_bam.py which will output a reference genome that matches the input BAM if it's possible to do so.
